File: rehovot-ads-packages/android/src/main/java/com/rehovottech/levelplay/UnityLevelPlay.kt
package com.rehovottech.capacitorads

// Native facade placeholder. Real SDK wiring will live here when each backend is added.
class CapacitorAds(
    private val logger: Logger,
) {
    public fun initialize(options: InitializeRequest) {
        logger.setDebugEnabled(options.debug)
        logger.d("TODO: initialize native ad SDK")
        logger.d("initialize options = $options")
    }

    public fun showBanner(options: BannerRequest) {
        logger.d("TODO: show banner")
        logger.d("banner request = $options")
    }

    public fun hideBanner() {
        logger.d("TODO: hide banner")
    }

    public fun destroyBanner() {
        logger.d("TODO: destroy banner")
    }

    public fun showInterstitial(options: InterstitialRequest) {
        logger.d("TODO: show interstitial")
        logger.d("interstitial request = $options")
    }

    public fun showRewarded(options: RewardedRequest): RewardResultPayload {
        logger.d("TODO: show rewarded")
        logger.d("rewarded request = $options")
        return RewardResultPayload(
            success = false,
            completed = false,
            message = "Native rewarded flow is not implemented yet.",
            placementId = options.placementId,
            rewardAmount = options.rewardAmount,
            rewardCurrency = options.rewardCurrency,
        )
    }

    public fun destroy() {
        logger.d("TODO: destroy native ad SDK")
    }
}

internal data class InitializeRequest(
    val appKey: String? = null,
    val testMode: Boolean = false,
    val debug: Boolean = false,
    val autoStart: Boolean = true,
)

internal data class BannerRequest(
    val placementId: String? = null,
    val position: String? = null,
)

internal data class InterstitialRequest(
    val placementId: String? = null,
    val timeoutMs: Int? = null,
)

internal data class RewardedRequest(
    val placementId: String? = null,
    val rewardAmount: Int? = null,
    val rewardCurrency: String? = null,
)

internal data class RewardResultPayload(
    val success: Boolean,
    val completed: Boolean,
    val message: String,
    val placementId: String? = null,
    val rewardAmount: Int? = null,
    val rewardCurrency: String? = null,
)

File: rehovot-ads-packages/ios/Plugin/UnityLevelPlay.swift
import Foundation

// Native facade placeholder. Real SDK wiring will live here when each backend is added.
final class CapacitorAds {
    private let logger: Logger

    init(logger: Logger) {
        self.logger = logger
    }

    func initialize(_ options: InitializeRequest) {
        logger.setDebugEnabled(options.debug)
        logger.debug("TODO: initialize native ad SDK")
        logger.debug("initialize options: \(options)")
    }

    func showBanner(_ options: BannerRequest) {
        logger.debug("TODO: show banner")
        logger.debug("banner request: \(options)")
    }

    func hideBanner() {
        logger.debug("TODO: hide banner")
    }

    func destroyBanner() {
        logger.debug("TODO: destroy banner")
    }

    func showInterstitial(_ options: InterstitialRequest) {
        logger.debug("TODO: show interstitial")
        logger.debug("interstitial request: \(options)")
    }

    func showRewarded(_ options: RewardedRequest) -> RewardResultPayload {
        logger.debug("TODO: show rewarded")
        logger.debug("rewarded request: \(options)")
        return RewardResultPayload(
            success: false,
            completed: false,
            message: "Native rewarded flow is not implemented yet.",
            placementId: options.placementId,
            rewardAmount: options.rewardAmount,
            rewardCurrency: options.rewardCurrency
        )
    }

    func destroy() {
        logger.debug("TODO: destroy native ad SDK")
    }
}

struct InitializeRequest: CustomStringConvertible {
    let appKey: String?
    let testMode: Bool
    let debug: Bool
    let autoStart: Bool

    var description: String {
        "InitializeRequest(appKey: \(String(describing: appKey)), testMode: \(testMode), debug: \(debug), autoStart: \(autoStart))"
    }
}

struct BannerRequest: CustomStringConvertible {
    let placementId: String?
    let position: String?

    var description: String {
        "BannerRequest(placementId: \(String(describing: placementId)), position: \(String(describing: position)))"
    }
}

struct InterstitialRequest: CustomStringConvertible {
    let placementId: String?
    let timeoutMs: Int?

    var description: String {
        "InterstitialRequest(placementId: \(String(describing: placementId)), timeoutMs: \(String(describing: timeoutMs)))"
    }
}

struct RewardedRequest: CustomStringConvertible {
    let placementId: String?
    let rewardAmount: Int?
    let rewardCurrency: String?

    var description: String {
        "RewardedRequest(placementId: \(String(describing: placementId)), rewardAmount: \(String(describing: rewardAmount)), rewardCurrency: \(String(describing: rewardCurrency)))"
    }
}

struct RewardResultPayload {
    let success: Bool
    let completed: Bool
    let message: String
    let placementId: String?
    let rewardAmount: Int?
    let rewardCurrency: String?
}
